### ISSUE TITLE
Change Domain and DAA Lists file format from JSON to YAML or other suitable format

### ISSUE THREAD

**Issue by SebastianZimmeck** [privacy-tech-lab/optmeowt-browser-extension#45]
As just discussed with @kalicki1, it may be worthwhile to also implement support for custom header and cookies. The idea is that we could re-use the functionality we already have (e.g., DAA cookie functionality per issue #42 and our own header implementation). There could be another JSON spec (or two JSON specs; one for cookies and one for headers) where users could enter custom do not sell cookies and headers of websites that implement their own custom cookie- or header-based solutions.

**Comment by SebastianZimmeck** [comment]
(cc'ing @davebaraka and @rgoldstein01)

**Comment by kalicki1** [comment]
**Recent Commit Notes**

The two recent commits were attempts at implementing the discussed custom cookies functionality. Commit 
d1b08f5 adds support for reading multiple custom cookie JSON files (namely the 3rd party DAA cookies and any custom cookies in another JSON) and setting them respectively. Since I had a hard time finding support for writing to JSON files from within a Chrome extension (online it seems people converge around using FS from Node.js, though I haven't tested Node in extensions that well), Commit 6880ab7 implements support for saving all custom cookies to the local storage, which is then pulled when an OptMeowt backup text file is requested for export. 

**Future Work on Issue #45**

In future updates, I will add the UI for modifying these custom cookies and headers, and allow a more generalized support for parsing what elements are saved to our backup settings files on when restoring old user settings on import.

**Comment by kalicki1** [comment]
As discussed in our meeting with @SebastianZimmeck just now, we can leave out the custom cookie export functionality I added in the last commit. This can be something we add back in in the future when we have more time. 

For now, I will look into changing our JSON cookie list files to YAML, or a related file format, since it supports comments for lists that might need them. At the moment, we envision that advanced users who have their own custom cookies lists can place them in the respective YAML files themselves. A simple extension refresh should then place these cookies. This way we will not have this functionality within the UI and can keep this issue simple. I can just add a note from within the options page that this functionality is available to advanced users.

**Comment by kalicki1** [comment]
**1st Commit Notes (`path` support & removed dev functions)** 
The first of the two commits removed some of the export to files functionality for users' custom cookies I had begun to implement. These were in development so it is not a huge loss at the moment. The same commit also added a check for a `path` key when it parses a document for cookies (regardless of doc type format). This way we can handle the custom `path` edge cases that came up with the DAA tool in issue #50. 

**2nd Commit Notes (YAML support)**
The second commit added full support for YAML files in place of JSON storage documents. Since we are not using these documents to send automated data between computers, we figured the easy usability and the ability to add comments into YAML files was worth the time investment to make it work.

I spent a decent length of time scouring online for a JS YAML library we can easily implement for our Chrome extension use case, though the major players, such as [js-yaml](https://github.com/nodeca/js-yaml) from [yaml.org](https://yaml.org/), all seem to built with Node.JS in mind. Though I spent some time trying to somehow make these popular libraries compatible with our extension by adding full Node support, using [Browserify](http://browserify.org/), or by modifying the libraries, I found [yaml](https://github.com/eemeli/yaml) turned out to have a version that was compatible with non-Node.JS projects. This is what I included in OptMeowt's final libraries.
